rpmlb/builder/koji.py

```diff
@@ -0,0 +1,173 @@
+"""Builder interface for the Koji build service."""
+
+import re
+from pathlib import Path
+from typing import Iterator, Optional, Sequence
+
+from ..utils import run_cmd, run_cmd_with_capture
+from ..work import Work
+from .base import BaseBuilder
+
+
+class KojiBuilder(BaseBuilder):
+    """Builder interface for the Koji build service."""
+
+    DEFAULT_TARGET_FORMAT = 'sclo{epel}-{collection}-rh-el{epel}'
+
+    def __init__(self):
+        """Initialize the slots"""
+
+        self.collection = None  #: The collection to build
+        self.epel = None  #: Version of EPEL to build for
+        self.profile = None  #: The koji configuration profile to use
+        #: The format string for construction of the actual template name
+        self.target_format = type(self).DEFAULT_TARGET_FORMAT
+
+    @property
+    def base_command(self) -> Sequence[str]:
+        """Basis of the command for build service interaction."""
+
+        binary = 'koji'
+
+        if self.profile is None:
+            return [binary]
+        else:
+            return [binary, '--profile', self.profile]
+
+    @property
+    def target(self) -> str:
+        """The koji target to build into."""
+
+        return self.target_format.format(
+            collection=self.collection,
+            epel=self.epel,
+        )
+
+    def run(self, work: Work, **kwargs):
```

Comment:
I agree, and this override is a currently necessary "hack".

The problem is that I need an access to `Work` object and the CLI arguments (`**kwargs`) to set the builder's parameters. The CLI parameters are passed to `build` method and I could set them there (which means setting them over for each package!), but I have no good way to access the `Work` object. The `prepare` cannot be used, as it is skipped on resume.

The probably cleanest way to remedy this would probably be extending the `__init__` method to have access to `Work` and the rest of the CLI arguments. However, this would mean changing all existing builders and the way the builder is loaded and constructed. That change is bit out of scope of just `KojiBuilder`…

Suggestions on how to do this better are welcome 😉. If nothing will pop up, I will prepare a new PR with the `__init__` changes suggested above.